postgresql/goose/migration.go

```diff
@@ -0,0 +1,61 @@
+package goose
+
+import (
+	"context"
+	"database/sql"
+	"runtime"
+
+	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/stdlib"
+	"github.com/pressly/goose/v3"
+
+	"github.com/Scalingo/go-utils/errors/v2"
+)
+
+type PgxContextMigration func(context.Context, *pgx.Conn) error
+
+// AddPgxContextMigration registers a migration in the goose migrations list using a pgx.Conn instead of a sql.DB
+// It can only be used with a pgx driver
+func AddPgxContextMigration(upMigration PgxContextMigration, downMigration PgxContextMigration) error {
```

Comment:
nitpick: that's just me being picky but the ideal would be that this function takes a context in argument so that we can use it in `errors.New`.

Reply:
Oh nice idea!